packages/flutter_tools/lib/src/fuchsia/fuchsia_sdk.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/common.dart';
+import '../base/context.dart';
+import '../base/process.dart';
+
+/// The fuchsia SDK.
+FuchsiaSdk get fuchsiaSdk => context[FuchsiaSdk];
+
+/// The fuchsia SDK shell commands.
+///
+/// This workflow assumes development within the fuchsia source tree, providing
+/// the `fx` command-line tool.
+class FuchsiaSdk {
+  /// Invokes the `netaddr` command and return the output.
+  ///
+  /// This returns the network address of an attached fuchsia device.
+  String netaddr() {
```

Comment:
Run async and return `Stream<Uint8List>`[1]

[1] I know that the declared type of `stdout` is `Stream<List<int>>`, but under the covers, it's guaranteed to be a `Stream<Uint8List>`, and while I understand why it isn't that way[2], it's always nagged at me that the API isn't `Stream<Uint8List>` 😄.  So we may as well cast and expose the modern type here. <span><</span>/soapbox>

[2] afaiui, it's because the `Process` API predates the `Uint8List` type, and we don't want to make the API breaking change at this point.

Reply:
Is there an advantage that `runDetached` has over `runAsync`?  (I'm assuming these are the underlying commands to use). Also attempting to replace runSync (or runAsync) with runDetached/start is returning a process with a `null` stdout and stderr. still looking into why that is

Reply:
There's a bug to simplify the number of those API methods... I'd stick with runAsync